File: EasyWechatWeb/EasyWeChatModels/Entitys/Screen/Report.cs
using SqlSugar;

namespace EasyWeChatModels.Entitys;

/// <summary>
/// 报表实体类
/// </summary>
[SugarTable("Report", "报表表")]
public class Report
{
    /// <summary>
    /// 报表ID（主键）
    /// </summary>
    [SugarColumn(IsPrimaryKey = true)]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// 报表名称
    /// </summary>
    [SugarColumn(Length = 100, ColumnDescription = "报表名称")]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 分类
    /// </summary>
    [SugarColumn(Length = 50, ColumnDescription = "分类")]
    public string Category { get; set; } = string.Empty;

    /// <summary>
    /// 数据源ID
    /// </summary>
    [SugarColumn(ColumnDescription = "数据源ID")]
    public Guid DatasourceId { get; set; }

    /// <summary>
    /// SQL查询语句
    /// </summary>
    [SugarColumn(ColumnDataType = "longtext", ColumnDescription = "SQL查询")]
    public string SqlQuery { get; set; } = string.Empty;

    /// <summary>
    /// 是否显示图表
    /// </summary>
    [SugarColumn(ColumnDescription = "显示图表")]
    public bool ShowChart { get; set; } = true;

    /// <summary>
    /// 是否显示数据表格
    /// </summary>
    [SugarColumn(ColumnDescription = "显示表格")]
    public bool ShowTable { get; set; } = true;

    /// <summary>
    /// 图表类型 (bar/line/pie)
    /// </summary>
    [SugarColumn(Length = 20, ColumnDescription = "图表类型")]
    public string ChartType { get; set; } = "bar";

    /// <summary>
    /// X轴字段名
    /// </summary>
    [SugarColumn(Length = 50, IsNullable = true, ColumnDescription = "X轴字段")]
    public string? XAxisField { get; set; }

    /// <summary>
    /// Y轴字段名
    /// </summary>
    [SugarColumn(Length = 50, IsNullable = true, ColumnDescription = "Y轴字段")]
    public string? YAxisField { get; set; }

    /// <summary>
    /// 聚合类型 (sum/count/avg)
    /// </summary>
    [SugarColumn(Length = 20, ColumnDescription = "聚合类型")]
    public string Aggregation { get; set; } = "sum";

    /// <summary>
    /// 是否自动生成列配置
    /// </summary>
    [SugarColumn(ColumnDescription = "自动列配置")]
    public bool AutoColumns { get; set; } = true;

    /// <summary>
    /// 列模板ID
    /// </summary>
    [SugarColumn(IsNullable = true, ColumnDescription = "列模板ID")]
    public Guid? ColumnTemplateId { get; set; }

    /// <summary>
    /// 列配置JSON
    /// </summary>
    [SugarColumn(ColumnDataType = "longtext", IsNullable = true, ColumnDescription = "列配置JSON")]
    public string? ColumnConfigs { get; set; }

    /// <summary>
    /// 创建人ID
    /// </summary>
    [SugarColumn(ColumnDescription = "创建人ID")]
    public Guid CreatorId { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    [SugarColumn(ColumnDescription = "创建时间")]
    public DateTime CreateTime { get; set; } = DateTime.Now;

    /// <summary>
    /// 更新时间
    /// </summary>
    [SugarColumn(IsNullable = true, ColumnDescription = "更新时间")]
    public DateTime? UpdateTime { get; set; }
}

File: PCWeb/src/types/report.ts
// src/types/report.ts

import type { ColumnConfig, DetectedColumn } from './columnTemplate'

/**
 * 图表类型
 */
export type ChartType = 'bar' | 'line' | 'pie'

/**
 * 聚合类型
 */
export type AggregationType = 'sum' | 'count' | 'avg'

/**
 * 报表信息
 */
export interface Report {
  id: string
  name: string
  category: string
  datasourceId: string
  datasourceName?: string
  sqlQuery: string
  // 显示配置
  showChart: boolean
  showTable: boolean
  chartType: ChartType
  xAxisField?: string
  yAxisField?: string
  aggregation: AggregationType
  // 列配置
  autoColumns?: boolean
  columnTemplateId?: string
  columnConfigs?: ColumnConfig[]
  // 审计信息
  creator?: string
  createTime?: string
  updateTime?: string
}

/**
 * 报表分类
 */
export interface ReportCategory {
  id: string
  name: string
  icon?: string
}

/**
 * 报表列表查询参数
 */
export interface ReportListParams {
  pageIndex?: number
  pageSize?: number
  name?: string
  category?: string
}

/**
 * 创建报表参数
 */
export interface CreateReportParams {
  name: string
  category: string
  datasourceId: string
  sqlQuery: string
  showChart?: boolean
  showTable?: boolean
  chartType?: ChartType
  xAxisField?: string
  yAxisField?: string
  aggregation?: AggregationType
  autoColumns?: boolean
  columnTemplateId?: string
  columnConfigs?: ColumnConfig[]
}

/**
 * 更新报表参数
 */
export interface UpdateReportParams {
  id: string
  name?: string
  category?: string
  datasourceId?: string
  sqlQuery?: string
  showChart?: boolean
  showTable?: boolean
  chartType?: string
  xAxisField?: string
  yAxisField?: string
  aggregation?: string
  autoColumns?: boolean
  columnTemplateId?: string
  columnConfigs?: ColumnConfig[]
}

/**
 * 报表预览参数
 */
export interface PreviewParams {
  datasourceId: string
  sqlQuery: string
  chartType?: ChartType
  xAxisField?: string
  yAxisField?: string
  aggregation?: AggregationType
}

/**
 * 图表数据项
 */
export interface ChartDataItem {
  name: string
  value: number
}

/**
 * 报表统计摘要
 */
export interface ReportSummary {
  total: number
  count: number
  avg: number
}

/**
 * 报表预览结果
 */
export interface PreviewResult {
  chartData: ChartDataItem[]
  tableData: Record<string, any>[]
  detectedColumns: DetectedColumn[]
  summary: ReportSummary
}